src/main/java/uk/gov/hmcts/divorce/noticeofchange/event/SystemApplyNoticeOfChange.java

```diff
@@ -0,0 +1,135 @@
+
+        CaseData responseData = objectMapper.convertValue(response.getData(), CaseData.class);
+
+        updateChangeOfRepresentation(details, responseData, sysUserToken, s2sToken);
+
+        details.setData(responseData);
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(details.getData())
+            .state(details.getState())
+            .build();
+    }
+
+    private void updateChangeOfRepresentation(CaseDetails<CaseData, State> details,
+                                              CaseData responseData, String sysUserToken, String s2sToken) {
+        var caseData = details.getData();
+        var changeOrganisationRequest = caseData.getChangeOrganisationRequestField();
+        var loggedInUserEmail = changeOrganisationRequest.getCreatedBy().toLowerCase();
+        var applicant1Solicitor = responseData.getApplicant1().getSolicitor();
+        var applicant2Solicitor = responseData.getApplicant2().getSolicitor();
+        var organisationId = changeOrganisationRequest.getOrganisationToAdd().getOrganisationId();
+
+        List<ProfessionalUser> organisationUsers =
+                Optional.ofNullable(organisationClient.getOrganisationUsers(sysUserToken, s2sToken, organisationId))
+                        .map(FindUsersByOrganisationResponse::getUsers)
+                        .orElse(Collections.emptyList());
+
+        final var nocRequestingUser = organisationUsers.stream()
+                .filter(user -> user.getEmail().equalsIgnoreCase(loggedInUserEmail))
+                .findFirst()
+                .orElseThrow();
+
+        String nocSolicitorOrgName = organisationClient
+                .getOrganisationByUserId(sysUserToken, s2sToken, nocRequestingUser.getUserIdentifier())
+                .getName();
+
+        if (APPLICANT_1_SOLICITOR.getRole().equals(changeOrganisationRequest.getCaseRoleId().getRole())) {
+            updateOrgPolicyAndSolicitorDetails(applicant1Solicitor, nocSolicitorOrgName,
+                    nocRequestingUser, loggedInUserEmail, organisationId);
+        } else {
+            updateOrgPolicyAndSolicitorDetails(applicant2Solicitor, nocSolicitorOrgName,
+                    nocRequestingUser, loggedInUserEmail, organisationId);
+        }
+    }
+
+    private static void updateOrgPolicyAndSolicitorDetails(Solicitor applicantSolicitor, String nocSolicitorOrgName,
+                                                           ProfessionalUser nocRequestingUser, String loggedInUserEmail,
+                                                           String organisationId) {
+
+        OrganisationPolicy<UserRole> updatedOrganisationPolicy = OrganisationPolicy
+                .<UserRole>builder()
+                .organisation(Organisation
+                        .builder()
+                        .organisationId(organisationId)
+                        .organisationName(nocSolicitorOrgName)
+                        .build())
+                .build();
```

Comment:
I'm not sure we need to change this, as it looks like [AAC should do it](https://github.com/hmcts/aac-manage-case-assignment/blob/d071d8c7920450ee494af33f8e1d89f9e03b6771/src/main/java/uk/gov/hmcts/reform/managecase/service/noc/ApplyNoCDecisionService.java#L162)? I also can't see it in [probate's NoC PR](https://github.com/hmcts/probate-back-office/pull/2347/files#diff-80daf4bbf4d83a3099bb70dc6a6ef5b0673193811068f8b74fd3de2752519dc7R46-R67)